examples/js/ThreeLoader.js

```diff
@@ -0,0 +1,71 @@
+/* global itowns, Promise */
+var manager = new itowns.THREE.LoadingManager();
+
+/**
+ * Load a script asynchronously
+ * @param {String} uri URI of the script to load.
+ * @returns {Promise} A promise resolved when the script is loaded.
+ */
+function loadScriptAsync(uri) {
+    return new Promise(function p(resolve) {
+        var firstScriptTag = document.getElementsByTagName('script')[0];
+        var tag = document.createElement('script');
+        tag.src = uri;
+        tag.async = true;
+        tag.onload = function r() {
+            resolve();
+        };
+        firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+    });
+}
+
+/**
+ * Load script asynchronously from ThreeJs example loaders
+ *
+ * @param {String} format Format to be loader. Example: Collada will return ColladaLoader
+ * @returns {Promise} A promise with the ThreeJS Loader
+ */
+function getThreeJsLoader(format) {
+    var res;
+    var promise = new Promise(function p(resolve) {
+        res = resolve;
+    });
+    // eslint-disable-next-line no-undef
+    THREE = itowns.THREE;
+    loadScriptAsync('https://cdn.rawgit.com/mrdoob/three.js/r' + itowns.THREE.REVISION + '/examples/js/loaders/' + format + 'Loader.js')
+    .then(function createLoader() {
+        res(new itowns.THREE[format + 'Loader'](manager));
+    });
+    return promise;
+}
+
+/**
+ * Load a ressource asynchronously using an already created loader
+ * @param {Loader} loader Loader from ThreeJs example loaders
+ * @param {String} url URL of the ressource
+ * @returns {Promise} A promise resolved when the ressource is loaded.
+ */
+function useThreeJsLoader(loader, url) {
+    var res;
```

Comment:
As you often use this structure you can declare it only once

```js
function defer() {
    const deferedPromise = {};
    deferedPromise.promise = new Promise((resolve, reject) => {
        deferedPromise.resolve = resolve;
        deferedPromise.reject = reject;
    });
    return deferedPromise;
}
```

Reply:
Thank's, nice refactoring idea !!
It removes duplicate code ;)